packages/rum-core/src/performance-monitoring/transaction-service.js

```diff
@@ -153,6 +164,8 @@ class TransactionService {
       }
     }
 
+    this.ensureRespInterval()
```

Comment:
It seems like instead of calling this function always and checking for `checkBrowserResponsiveness`, we can just move this inside else block of if `(tr.type === PAGE_LOAD)` and get rid of some code from this function.

Reply:
IMO, we should keep the special cases for the transaction type in one place as much as possible. `ensureRespInterval` function only needs to know about `checkBrowserResponsiveness` and that's all the information it needs. if we have a check in that function we need to remember to update that code path as well every time we change something.